src/event.rs

```diff
@@ -71,7 +70,7 @@ pub enum Event {
     RepeatedHttpSessions(RepeatedHttpSessions),
 
     /// An HTTP connection to a Tor exit node.
-    TorConnection(TorConnection),
+    TorConnection(Box<TorConnection>),
 
     /// DGA (Domain Generation Algorithm) generated hostname in HTTP request message
     DomainGenerationAlgorithm(Box<DomainGenerationAlgorithm>),
```

Comment:
Could you try below with corresponding changes in line 1043 and 1054?
```suggestion
    TorConnection(TorConnection),

    /// DGA (Domain Generation Algorithm) generated hostname in HTTP request message
    DomainGenerationAlgorithm(DomainGenerationAlgorithm),
```